eZ/Publish/Core/IO/IOMetadataHandler/LegacyDFSCluster.php

```diff
@@ -47,17 +47,21 @@ public function __construct(Connection $connection, UrlDecorator $urlDecorator =
     /**
      * Inserts a new reference to file $spiBinaryFileId.
      *
-     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException If a file $spiBinaryFileId already exists
+     * @since 6.8 The mtime of the $binaryFileCreateStruct must be a DateTime, as specified in the struct doc.
      *
-     * @param string  $spiBinaryFileId
+     * @param SPIBinaryFileCreateStruct $binaryFileCreateStruct
      *
-     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException if an error occurs creating the record
+     * @throws InvalidArgumentException if the $binaryFileCreateStruct is invalid
      * @throws RuntimeException if a DBAL error occurs
      *
      * @return \eZ\Publish\SPI\IO\BinaryFile
      */
     public function create(SPIBinaryFileCreateStruct $binaryFileCreateStruct)
     {
+        if (!($binaryFileCreateStruct->mtime instanceof DateTime)) {
```

Comment:
Possible followup improvment, but you can pherhaps do it here already:
- Update doc and code here to allow [DateTimeInterface](http://php.net/manual/en/class.datetimeinterface.php) which was added in PHP 5.5 and up